Microsoft.HealthVault.Fhir/Transformers/MedicationStatementToHealthVault.cs

```diff
@@ -0,0 +1,92 @@
+        {
+            var fhirMedication = ExtractEmbeddedMedication(medicationStatement);
+
+            var hvMedication = fhirMedication.ToHealthVault() as HVMedication;
+
+            if (medicationStatement.Dosage.Any())
+            {
+                var dosage = medicationStatement.Dosage.First();
+
+                switch (dosage.Dose)
+                {
+                    case null:
+                        break;
+                    case SimpleQuantity doseQuantity:
+                        var dose = new GeneralMeasurement { };
+                        dose.Structured.Add(new StructuredMeasurement
+                        {
+                            Value = (double)doseQuantity.Value,
+                            Units = CodeToHealthVaultHelper.CreateCodableValueFromQuantityValues(doseQuantity.System, doseQuantity.Code, doseQuantity.Unit)
+                        });
+                        hvMedication.Dose = dose;
+                        break;
+                    case Range doseRange:
+                        throw new NotImplementedException();
+                }
+
+                Timing.RepeatComponent repeat = dosage.Timing?.Repeat;
+                if (repeat?.Period != null && repeat?.PeriodUnit != null)
+                {
+                    var frequency = new GeneralMeasurement { };
+                    frequency.Structured.Add(new StructuredMeasurement
+                    {
+                        Value = (double)repeat.Period,
+                        Units = CodeToHealthVaultHelper.GetRecurrenceIntervalFromPeriodUnit(repeat.PeriodUnit.Value)
+                    });
+                    hvMedication.Frequency = frequency;
+                }
+
+                var route = dosage.Route.ToCodableValue();//.GetCodableValue();
+                hvMedication.Route = route;
+
+            }
+            return hvMedication;
+        }
+
+        private static FhirMedication ExtractEmbeddedMedication(MedicationStatement medicationStatement)
+        {
+            switch (medicationStatement.Medication)
+            {
+                case ResourceReference medicationReference:
+                    if (!medicationReference.IsContainedReference)
+                    {
+                        throw new NotImplementedException();
+                    }
+                    var fhirMedication = medicationStatement.Contained.First(domainResource
+                             => medicationReference.Matches(domainResource.GetContainerReference())) as FhirMedication;
+                    return fhirMedication ?? new FhirMedication();
+                case CodeableConcept medicationCodeableConcept:
+                    return new FhirMedication
+                    {
+                        Code = medicationCodeableConcept
+                    };
+                default:
+                    throw new NotImplementedException();
+            }
```

Comment:
This function  looks like it is exactly the same as in MedicationRequestToHealthVault.cs Please refactor and put on a common place, repeated code is never a good idea as you duplicate bugs across several copies of the same code